ruby/src/IceRuby/Communicator.cpp

```diff
@@ -41,12 +41,22 @@ IceRuby_Communicator_mark(Ice::CommunicatorPtr* p)
 }
 
 extern "C" void
-IceRuby_Communicator_free(Ice::CommunicatorPtr* p)
+IceRuby_Communicator_free(void* p)
 {
-    assert(p);
-    delete p;
+    auto communicator = static_cast<Ice::CommunicatorPtr*>(p);
+    delete communicator;
```

Comment:
I prefer to use the same style for all this free methods `delete static_cast<Ice::ConnectionPtr*>(p);` l